cvmfs/sync_item.h

```diff
@@ -282,6 +288,8 @@ class SyncItem {
   bool opaque_;                       /**< SyncUnion marked this as opaque dir*/
   bool masked_hardlink_;              /**< SyncUnion masked out the linkcount */
   bool has_catalog_marker_;           /**< directory containing .cvmfscatalog */
+  bool is_path_present_;
```

Comment:
We should be able to fix the issue without adding any new members to `SyncItem`.  I think it is sufficient to path `CheckCatalogMarker()`